src/api/report/reportRepository.ts

```diff
@@ -564,10 +570,11 @@ export const reportRepository = {
               timeToReachThreshold.full.push(diff);
             }
             threshold = 4;
+            dealRest = allocationUsed - BigInt(allocation);
+            if (dealRest === 0n) dealIdx++;
```

Comment:
This will just jump over some deals completely. Example:
Allocations: `[100, 100]`
Deals: `[50, 50, 50, 50]`

dealIdx=0 will reach thresholds 0 and 1. dealIdx=1 will reach thresholds 2 and 3 and get inside this if. Because allocationUsed = 100 and allocation = 100, dealRest will be 0, so we increment dealIdx to 2. We go to the next iteration of the loop, which will increment the dealIdx to 3 before execution. So we didn't do anything with dealIdx=2, just jumped over it.

Please coordinate with @neti-mateusz-mazur (maybe a pair-programming session?) and get a proper fix for this.

Reply:
Actually, scratch that, I missed the `break`. Should work fine, but please refactor it so that it's testable and write some tests for this logic. Cover both deals matching allocations exactly and "spilling over".